### DISCUSSION TITLE
Sender Address Column

### DISCUSSION THREAD

**Discussion by andybrice** [pimalaya/himalaya#400] [Ideas]
I've made myself a fork of Himalaya with a separate sender address column, like so:

```
ID   │FLAGS │SUBJECT                         │SENDER       │ADDRESS                │DATE                
1005 │✷     │Lorem ipsum dolor sit amet      │Porta Sapien │porta@example.com      │2022-07-19 11:47:48 
1004 │✷     │Consectetur adipiscing elit     │Elementum    │lacinia@example.com    │2022-07-19 11:39:48 
1003 │      │Nulla nec ligula                │Vel Tortor   │vel.tortor@example.com │2022-07-18 10:14:16 
1002 │      │Mauris efficitur ex vitae justo │Finibus      │finibus@example.com    │2022-07-18 10:08:14 
1001 │✷     │Viverra Bibendum                │Justo Luctus │admin@example.com      │2022-07-18 10:05:44 
```
(I did this mostly because I want the JSON output for a GUI.)

Do you have any interest in me trying to incorporate this back into Himalaya as a configuration option?

**Comment by soywod** [comment]
There is a related issue that aims to make output (table and JSON) customizable, see https://github.com/soywod/himalaya/issues/140. It is not a priority but it would be a very nice feature to merge, and it look very close to what you want to do.

About the GUI, in which language do you plan to write it? Because Himalaya is slowly shifting towards a generic library of email management rather than a CLI (this CLI will become the first consumer of the lib). If you use the same language as the lib (Rust) you could build native interface.

Last point, Himalaya will receive financial support very soon (see https://github.com/soywod/himalaya/discussions/399). Would you be interested in receiving a part of this funds to develop this feature (the customizable output and/or even the GUI)?

**Reply by andybrice** [reply]
I've been trying out [Tauri](https://tauri.app/) for GUIs. That Rust connection is what led me to Himalaya.

Okay, I'll bear #140 in mind so that my own work helps move towards it.

In theory I would certainly be interested in doing that as paid work! Though realistically I'm not sure I have the skills yet. I'm especially new to Rust.

**Reply by soywod** [reply]
> I've been trying out [Tauri](https://tauri.app/) for GUIs. That Rust connection is what led me to Himalaya.

It's funny because I discovered this lib not so long time ago and I thought about giving it a shot!

> Okay, I'll bear #140 in mind so that my own work helps move towards it.

Alright, I will keep it for you.

> In theory I would certainly be interested in doing that as paid work! Though realistically I'm not sure I have the skills yet. I'm especially new to Rust.

I learnt Rust with this project, so it's fine :) good to hear! For now I'm busy moving away from GitHub and separating the lib from the CLI. We need those 2 tasks solved before starting anything, because it is a mess right now. Everything should be ready for September, I let you know. Thank you!
